src/cargo/sources/registry/index.rs

```diff
@@ -189,11 +189,34 @@ enum MaybeIndexSummary {
 /// from a line from a raw index file, or a JSON blob from on-disk index cache.
 ///
 /// In addition to a full [`Summary`], we have information on whether it is `yanked`.
-pub struct IndexSummary {
-    pub summary: Summary,
-    pub yanked: bool,
-    /// Schema version, see [`IndexPackage::v`].
-    v: u32,
+#[derive(Clone, Debug)]
+pub enum IndexSummary {
+    /// Available for consideration
+    Candidate(Summary),
+    /// Yanked within its registry
+    Yanked(Summary),
+    /// From a newer schema version and is likely incomplete or inaccurate
+    Unsupported(Summary, u32),
+}
```

Comment:
Is there a reason you left off the error case from my branch?

Reply:
`?` can be used in functions that return `CargoResult<IndexSummary>`. It cannot be used in functions and return `IndexSummary`, even if it has an error variant. (try trait is not stable.) If `IndexSummary` has an error variant then `CargoResult<IndexSummary>` have two different states with the same semantic meaning.

Also it seems like the kind of thing that would be easy to do as a re-factor once all the infrastructure is set up. So it seems like kind decision that could easily be undone.